lib/core/crypto/dart_esr/src/signing_request_manager.dart

```diff
@@ -783,3 +789,41 @@ class SigningRequestUtils {
     return ESRConstants.ChainIdLookup[ChainName.RESERVED];
   }
 }
+
+extension HyphaSigningRequestManager on SigningRequestManager {
+  /// Convert the ESR standard's "chain_id" to our Network
+  /// Both signify a unique identifier for a chain
+  /// But we support fewer chains than ESR does.
+  /// Networks maps to remote config which contains the server node URLs we need
+  /// to use in order to access supported chains.
+  /// Returns: Network
+  /// throws: unsupported network when the chainID can't be parsed.
+  ///
+  static Networks resolveNetwork(List<dynamic> chainId) {
```

Comment:
dynamic -> do we know the type here?

Reply:
No - unfortunately this is a list that's mixed... coming from the EOSJS and ESR libraries. 

it can be [String, String] or [String, int]